p2p/host.go

```diff
@@ -160,32 +159,40 @@ func (h *host) ping(peer peer.ID) error {
 	return h.dht.Ping(h.ctx, peer)
 }
 
-// send sends a non-status message to a specific peer
+// send writes the given message to the outbound message stream for the given
+// peer (gets the already opened outbound message stream or opens a new one).
 func (h *host) send(p peer.ID, msg Message) (err error) {
 
-	// create new stream with host protocol id
-	s, err := h.h.NewStream(h.ctx, p, h.protocolId)
-	if err != nil {
-		return err
-	}
+	// get outbound stream for given peer
+	s := h.getStream(p)
 
-	log.Trace(
-		"Opened stream",
-		"host", h.id(),
-		"peer", p,
-		"protocol", s.Protocol(),
-	)
+	// check if stream needs to be opened
+	if s == nil {
 
-	encMsg, err := msg.Encode()
-	if err != nil {
-		return err
+		// open outbound stream with host protocol id
+		s, err = h.h.NewStream(h.ctx, p, h.protocolId)
+		if err != nil {
+			return err
+		}
+
+		log.Trace(
+			"Opened stream",
+			"host", h.id(),
+			"peer", p,
+			"protocol", s.Protocol(),
+		)
 	}
 
-	_, err = s.Write(common.Uint16ToBytes(uint16(len(encMsg)))[0:1])
+	encMsg, err := msg.Encode()
 	if err != nil {
 		return err
 	}
 
+	// _, err = s.Write(common.Uint16ToBytes(uint16(len(encMsg)))[0:1])
+	// if err != nil {
+	// 	return err
+	// }
```

Comment:
Can this be removed?

Reply:
I was going to remove it when taking a look at #484 but I can remove now.

Reply:
done